pandas/core/indexes/datetimes.py

```diff
@@ -470,6 +470,14 @@ def _formatter_func(self):
     # --------------------------------------------------------------------
     # Set Operation Methods
 
+    def _is_compatible_with_other(self, other):
```

Comment:
again, this should match exactly or it gets converted to object. why is this also necessary?

Reply:
Assumed we wanted different tz datetimeindexes to be compatible since their union results in converting to UTC and unioning them. However if we do not want this behavior then I can default to the super method on this too.

Reply:
no we do not want this. Things must match exactly.